include/i2c.h

```diff
@@ -45,8 +45,9 @@ extern "C" {
 /** I2C Ultra Fast Speed */
 #define I2C_SPEED_ULTRA			(0x5)
 
-/** @cond INTERNAL_HIDDEN */
 #define I2C_SPEED_SHIFT			(1)
+#define I2C_SPEED_SET(speed)		((speed) << I2C_SPEED_SHIFT)
```

Comment:
It's customary to write _SET macros as
```
#define I2C_SPEED_SET(speed)		(((speed) << I2C_SPEED_SHIFT) & I2C_SPEED_MASK)
```
to avoid unintentional overwriting of neighboring bit-fields.